lib/content-types/utils.js

```diff
@@ -25,27 +25,22 @@ const attributes = (config) => {
 
   // map returns either true or a string; filter leaves us with just an array of strings
   const attrs = config.attributes.map(attr => {
-    if ((!attr.hasOwnProperty('name') || attr.name === '')) {
-      let ident = attr.type;
-
-      if (attr.hasOwnProperty('id')) {
-        ident = attr.id;
-      }
-
-      return `Attribute '${ident}' in content type '${config.name}' needs a name`;
-    }
-    if (typeof attr.name !== 'string') {
+    // if there is a name, it must be a string
+    if (attr.hasOwnProperty('name') && attr.name !== '' && typeof attr.name !== 'string') {
       return `Attribute names in content type '${config.name}' must be a string`;
     }
-    if (!attr.id || attr.id === '') {
-      return `Attribute '${attr.name}' in content type '${config.name}' must have an id`;
-    }
-    if (typeof attr.id !== 'string') {
+
+    // if there is an, it must be a string
+    if (attr.hasOwnProperty('id') && attr.id !== '' && typeof attr.id !== 'string') {
```

Comment:
IDs are required. My [comment](https://github.com/punchcard-cms/content-types/pull/72#discussion_r72886299) said that `name` wasn't strictly needed, but `type` and `id` are. I also don't think we _need_ to roll back the requirement on having `name`. 

Before you say "but `id` is generated from `name`, we should specifically not do this, as there are multiple human-readable names that can compile down to the same slug ID, and because ID is used for so many different important things in our forms, we want that to be explicit, the same way we want content type names to be explicit.